### DISCUSSION TITLE
The latest version of shamiko is not compatible with magisk V. 26.3

### DISCUSSION THREAD

**Discussion by llLegion** [community/community#71914] [Programming Help]
### Select Topic Area

Bug

#71914 

i got an update for shamiko and after 
I updated to version v0.7.4(188) Shamiko failed to install in Magisk.
then I re-installed the previous version shamiko v0.7.3-174 and it worked .
I tried something else by removing all the modules, it still didn't work.
then I tried disabling Zygisk and installing the latest version of Shamiko but it still didn't work.
It seems that the latest version of Shamiko is not yet compatible with Magisk v26.3.
and I still choose to stick with shamiko v0.7.3

![Screenshot_2023-10-24-08-21-22-726_com topjohnwu magisk](https://github.com/community/community/assets/76087392/181fe92d-cd63-4a00-9d41-56785d540e16)

#issue

**Comment by thezeeshanhassan** [comment]
It seems you've encountered an issue with the compatibility of Shamiko and Magisk. The information you provided suggests that the latest version of Shamiko (v0.7.4-188) may not be compatible with Magisk v26.3. If you're experiencing problems with the latest version, you can stick with Shamiko v0.7.3-174, which is working for you.

In such cases, it's common for developers to need some time to update their modules to be compatible with the latest version of Magisk. You may want to keep an eye on the official Magisk and Shamiko channels for updates, as developers often release patches and updates to address compatibility issues with newer versions of Magisk.

If you have any specific questions or need further assistance with this issue, you might want to visit the XDA Developers forums or the respective GitHub repository for Shamiko to see if there are discussions or issue reports related to this specific compatibility problem. Developers and the community can provide more detailed insights and potential workarounds.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @llLegion ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. 

If you're asking how to use the piece of software that you found in a repository on GitHub, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.

**Comment by akha666** [comment]
Any update regarding this bug?
The latest magisk is 26300, and Shamiko requires 26302+

**Reply by caspaas** [reply]
i feel like they used a canary build maybe?

**Reply by akha666** [reply]
I thought that.

**Reply by mathewseduardo** [reply]
Right now i update to magisk 26.4 and install shamiko 0.7.4. all is ok.

**Comment by hideki2k02** [comment]
Switch from the Magisk Stable app to Magisk Canary app, then change update channel and flash the Latest Canary version.

**Comment by mathewseduardo** [comment]
Upgrade to magisk official 26.4 (26400) and install shamiko 0.7.4

**Comment by swieder711** [comment]
Currently running Shamiko .7.5
I tried to install Shamiko 1.0 with Magisk 264002 and it FAILED.
Any suggestions?

**Comment by Razib092** [comment]
Shamiko new update install failed 
![Screenshot_2024-01-08-18-50-48-86_e5bea2b0b2afd3b7e5e09a4b877e73c2](https://github.com/community/community/assets/129007408/4f5f6404-0ac8-48e7-b1cb-895819e27125)
Any solution

**Reply by swieder711** [reply]
You need to be running Magisk 26404 or later

**Reply by Razib092** [reply]
But I'm already in latest version megisk, 26.4

**Reply by swieder711** [reply]
26404 is the canary release and it newer than 26.4

**Reply by Ostin69** [reply]
Same error, question why we need Canary build and not official Magisk 26400 ?
Because moving to canary is quite complicated in some cases

264004 has issues with AD Guard
https://github.com/AdguardTeam/adguardcert/issues/52

in that case Magisk Delta will be better
https://github.com/HuskyDG/magisk-files/releases

**they also fixed:**
[Module] Fix symlink is mistakely marked as whiteout for systemless deteting, which cause symlink fails to be added to system. This problem exists from the first day of kitsune and no one reports 😑, also there are not much modules that replace/add symlinks. Please update to this version to solve the symlink problem.

**Diffs to official Magisk**
[App] Added a new feature to install Magisk into /system partition for emulators that do not have a boot image.
[Zygisk] Changed the method of loading Zygisk to ptrace init implementation (required Android 8+). The implementation comes from [ZygiskNext](https://github.com/Dr-TSNG/ZygiskNext). This method does not need to change any property and is more compatible with some emulators that ignore ro.dalvik.vm.native.bridge property.
[General] Enhanced the support for mounting in pre-init for modules. This enables modules to modify the system before the init process starts, which can be useful for some advanced use cases.
[General] Implemented a new feature to support deleting file by using indicated whiteout character device, similar to overlayfs. This allows modules to delete files from the original system without actually modifying it.
[Zygisk] Added support for GrepheneOS Android 14, a privacy and security focused mobile OS with Android app compatibility. This is because the official Magisk does not support it.
[General] Restored support for devices with no selinux support, which was removed by the official Magisk. This allows users to use Magisk on devices that do not have selinux enabled in kernel.
[General] Change the mounting su location from /system/bin to /apex/com.android.runtime/bin. This is to reduce bind mount. For most apps (that don't change/hardcode PATH variable), this change should not break anything. An known app is Termux which hardcode PATH might fail to locate su, you can call /debug_ramdisk/su.
[Module] Cancel the writable exception for /system/etc/hosts. Please write to safely place /data/adb/modules/hosts/system/etc/hosts (built-in hosts) instead. App like Adaway might not work.

**Comment by MichalR1234** [comment]
Anyway could you please add support for Magisk Kitsune for shamiko?

I using Kitsune in 26404 version which is the same like Delta... 
![IMG_20240113_214750](https://github.com/community/community/assets/156303561/f607d737-0b22-460b-98bc-903b67925b92)

![IMG_20240113_215246](https://github.com/community/community/assets/156303561/a204a279-0d13-48cd-bdb4-c1895ed4426a)

**Reply by Thesola10** [reply]
This looks like a successful install to me, the Japanese text is a quote from the anime the name Shamiko is from.

**Reply by MichalR1234** [reply]
I don't think so because in version 0.7.5 it was subscribed other.

**Reply by MichalR1234** [reply]
![IMG_20240129_202255](https://github.com/community/community/assets/156303561/88ba3e24-c17e-4496-8f87-aa02980ace0c)

**Comment by tomtom8887-tech** [comment]
I installed Shamiko v1.0.1 on magiskdelta version 27001 and it failed.

**Comment by Lutss01** [comment]
Does Shamiko V1.1 support Android 14?
